src/nevermined/Assets.ts

```diff
@@ -275,6 +297,11 @@ export class Assets extends Instantiable {
                         rating: 0,
                         numVotes: 0
                     },
+                    additionalInformation: {
+                        customData: {
+                            nftContractAddress: nftTokenAddress
```

Comment:
Isn't the `nftTokenAddress` already present in the template conditions?
The sdk shouldn't have to rely on custom date to function correctly

Reply:
We can just add a method to the aavecredittemplate to get the contract address from the nftLock condition

Reply:
this will override the additionalInformation field supplied by the user when creating an nft which is not a good thing since the additionalnformation field is there for user provided custom data

Reply:
i have reverted this change and added a function to get the address from the service condition